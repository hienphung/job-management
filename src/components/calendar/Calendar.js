import React, {Component} from "react";
import {Text, TouchableOpacity, View} from "react-native";
import styles from "./styles";
import PropTypes from "prop-types";

export default class Calendar extends Component {
	static propTypes = {
		data: PropTypes.arrayOf(PropTypes.object)
	};


	constructor(props) {
		super(props);
	}

	renderDate(value, index) {
		const {onItemPress} = this.props;
		if (index !== this.props.selectedIndex) {
			return (
				<TouchableOpacity onPress={() => {
					onItemPress ? onItemPress(index) : null
				}}>
					<View style={styles.dateTextContainer}>
						<Text style={styles.dateText}>{value}</Text>
					</View>
				</TouchableOpacity>
			)
		}
		return (
			<View style={styles.dateTextSelectedContainer}>
				<Text style={styles.dateTextSelected}>{value}</Text>
			</View>
		)
	}

	render() {
		const {data} = this.props;
		if (data) {
			return (
				<View style={styles.container}>
					{
						data.map((value, index) => {
							return (
								<View style={styles.dateContainer} key={value.id}>
									<Text style={styles.shortWeekDateText}>{value.shortWeekDate}</Text>
									{this.renderDate(value.date, index)}
								</View>
							)
						})
					}
				</View>
			)
		}
		return null;
	}
}