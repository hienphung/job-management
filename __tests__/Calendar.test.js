import React from 'react';
import {shallow} from 'enzyme';
import Calendar from "../src/components/calendar/Calendar";

describe("Test Calendar Component", () => {
	it("Should render correctly", () => {
		const component = shallow(<Calendar/>);
		expect(component).toMatchSnapshot();
	})
});