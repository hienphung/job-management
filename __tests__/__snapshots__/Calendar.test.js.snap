// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calendar Component Should render correctly 1`] = `ShallowWrapper {}`;
